src/main/java/org/terasology/launcher/config/ConfigManager.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.config;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import javafx.concurrent.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.launcher.packages.Package;
+import org.terasology.launcher.util.LauncherDirectoryUtils;
+import org.terasology.launcher.util.OperatingSystem;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+public class ConfigManager {
+    private static final Logger logger = LoggerFactory.getLogger(ConfigManager.class);
+    private static final String CONFIG_FILE = "config.json";
+
+    private final Path launcherDir;
+    private final Path configPath;
+    private final Gson gson;
+    private LauncherConfig config;
+    private final Service<LauncherConfig> loadService;
+    private final Service<Void> saveService;
+
+    public ConfigManager() {
```

Comment:
Do you envision the `ConfigManager` to be a singleton? In that case, the constructor should be private. As it is right now, one can easily create additional manager which might interfere with each other.

Reply:
That'd be helpful definitely. Will make it a proper singleton :+1: